src/main/java/seedu/address/model/stats/HistogramStat.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.model.stats;
+
+import org.jfree.chart.ChartFactory;
+import org.jfree.chart.JFreeChart;
+import org.jfree.data.statistics.HistogramDataset;
+
+import javafx.collections.ObservableList;
+import seedu.address.model.property.Buyer;
+import seedu.address.model.property.Property;
+
+public class HistogramStat implements Stat {
+    private static final int BIN_COUNT = 20;
+
+    private final ObservableList<Buyer> buyerList;
+    private final ObservableList<Property> propertyList;
+    private final boolean showBuyer;
+    private final boolean showProperty;
+    private final String titleArgs;
+
+    /**
+     * Constructor for price histograms.
+     * @param buyerList Currently viewable list of buyers.
+     * @param propertyList Currently viewable list of properties.
+     */
+    public HistogramStat(ObservableList<Buyer> buyerList, ObservableList<Property> propertyList,
+                         boolean showBuyer, boolean showProperty, String titleArgs) {
+        this.buyerList = buyerList;
+        this.propertyList = propertyList;
+        this.showBuyer = showBuyer;
+        this.showProperty = showProperty;
+        this.titleArgs = titleArgs;
+    }
+
+    /**
+     * Creates a histogram to displayed as a popup.
+     * Shows either buyers, properties or both.
+     *
+     * @return price histogram.
+     */
+    public JFreeChart create() {
+        double[] buyerPrices = buyerList.stream()
+                .mapToDouble(buyer -> (double) buyer.getMaxPrice().value).toArray();
+        double[] propertyPrices = propertyList.stream()
+                .mapToDouble(property -> (double) property.getPrice().value).toArray();
+
+        HistogramDataset dataset = new HistogramDataset();
+        if (showBuyer) {
+            dataset.addSeries("buyers", buyerPrices, BIN_COUNT);
```

Comment:
empty lists seem to cause exceptions here. 
Interestingly `stat` is not NULL within MainWindow but it just cannot draw here.
Exception in thread "JavaFX Application Thread" java.lang.IllegalArgumentException: Null or zero length 'values' argument.
	at org.jfree.data.statistics.HistogramDataset.getMinimum(HistogramDataset.java:198)
	at org.jfree.data.statistics.HistogramDataset.addSeries(HistogramDataset.java:113)
	at seedu.address.model.stats.HistogramStat.create(HistogramStat.java:48)
	at seedu.address.ui.StatWindow.show(StatWindow.java:60)
	at seedu.address.ui.MainWindow.handleStat(MainWindow.java:173)

Reply:
Alright sure, handling empty lists as explained in the comment above.